src/Lecture2.hs

```diff
@@ -70,7 +75,19 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt idx l
+  | idx < 0 = (Nothing, l)
+  | otherwise =
+    let (left, right) = splitAt idx l
+    in
+      case right of
+        [] -> (Nothing, left)
+        _ -> (Just (head right), left ++ tail right)
```

Comment:
You can pattern match on the explicit constructor of `right` here to avoid using unsafe functions `head` and `tail`:

```suggestion
        val : rest -> (Just val, left ++ rest)
```